backend/src/db/pool.ts

```diff
@@ -1,4 +1,9 @@
 import { Pool } from 'pg';
+import dotenv from 'dotenv';
+
+if (process.env.NODE_ENV === 'test') {
+    dotenv.config();
```

Comment:
Why is this required or when does it happen?
Maybe add some comments to explain this.

Reply:
I think I get it now after reading rest of the changes but it would still be good to give a short explanation in the comments